app/scripts/controllers/transactions/index.js

```diff
@@ -1151,13 +1151,28 @@ export default class TransactionController extends EventEmitter {
    *  params instead of allowing this method to generate them
    * @param options
    * @param options.estimatedBaseFee
+   * @param actionId
    * @returns {txMeta}
    */
   async createCancelTransaction(
     originalTxId,
     customGasSettings,
     { estimatedBaseFee } = {},
+    actionId,
   ) {
+    // In transaction is found for same action id, do not create a new cancel transaction.
+    if (actionId) {
+      const existingTxMeta =
+        this.txStateManager.getTransactionWithActionId(actionId);
+      if (existingTxMeta) {
+        // Method approveTransaction is confusing, but it actually approves and also submits / publishes the transaction
+        if (existingTxMeta.status === TRANSACTION_STATUSES.APPROVED) {
+          await this.approveTransaction(existingTxMeta.id);
```

Comment:
seems like this is added out of an abundance of caution but wanted to verify the state flow

https://github.com/MetaMask/metamask-extension/blob/2a27421e8e38612df47981faa6d59065cc810c2c/app/scripts/controllers/transactions/index.js#L1795-L1807

Reply:
@kumavis : PR is updated please check